### ISSUE TITLE
IndexOutOfBoundsException

### ISSUE THREAD

**Issue by halimpuckjava** [r2dbc/r2dbc-postgresql#136]
hi

i try to use r2dbc-postgresql with coroutines, so this is a snapshot of code:
```
suspend fun createConnection() = PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder().apply {
        host("host-name")
        port(port)
        database("database-name")
        username("username")
        password("password")
    }.build()).create().awaitFirst()

scope.launch {
       createStatement("select * from custom-table")
                    .execute()
                    .asFlow()
                    .map { it.map { row, _ -> row["employee"] }.asFlow() }  // this line throw exception
                    .flattenConcat()
                    .onEach { println(it) }
                    .catch { it.printStackTrace() }
                    .launchIn(scope)
}
```

in execution i see this exception:
java.lang.IndexOutOfBoundsException: index: 20201, length: 13 (expected: range(0, 16384))

**Comment by Squiry** [comment]
Related to #107 #91 #86

**Comment by mp911de** [comment]
That should be fixed with #116. Feel free to reopen if the problem persists.
